src/Controller/SecurityController.php

```diff
@@ -61,4 +66,27 @@ public function loginLink()
     {
         throw new \LogicException('This code should never be reached');
     }
+
+    #[Route('/resend-auth-code', name: 'resend_auth_code')]
+    public function resendAuthCode(EmailCodeGenerator $emailCodeGenerator, SMSCodeGenerator $smsCodeGenerator, EntityManagerInterface $em): RedirectResponse
+    {
+        /** @var User $user */
+        $user = $this->getUser();
+
+        if (!$user) {
+            return $this->redirectToRoute('re_main_login');
+        }
+
+        if (self::MFA_MAX_RETRIES === $user->getMfaRetryCount()) {
+            $this->addFlash('danger', 'mfa_maximum_retries_reach');
+
+            return $this->redirectToRoute('2fa_login');
+        }
+
+        User::MFA_METHOD_EMAIL === $user->getMfaMethod() ? $emailCodeGenerator->generateAndSend($user) : $smsCodeGenerator->generateAndSend($user);
+        $user->increaseMfaRetryCount();
+        $em->flush();
```

Comment:
What do you think about a separated service ? Not sure of the naming but
`$service->sendCode($user);` could be nice